added resol adapter to latest

Grizzelbee opened the pull request:
Added new adapter resol to latest repo. This adapter is derived from ioBroker.myVBus and is intended to be a replacement for this in the future.

Commit:
added resol adapter to latest

Apollon77 commented:
RE-CHECK!

GermanBluefox commented:
## Automated adapter checker

### [ioBroker.resol](https://github.com/Grizzelbee/ioBroker.resol)
[![Downloads](https://img.shields.io/npm/dm/iobroker.resol.svg)](https://www.npmjs.com/package/iobroker.resol) ![Number of Installations (latest)](http://iobroker.live/badges/resol-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.resol.png?downloads=true)](https://nodei.co/npm/iobroker.resol/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "resol" in latest repository


*Add comment "RE-CHECK!" to start check anew*

Apollon77 commented:
Hi, sorry that it took that long, here are some review comments:

* Please check https://github.com/Grizzelbee/ioBroker.resol/runs/4149640113?check_suite_focus=true#step:6:360
* If you add common.globalDependencies with admin>=4.0.9 you can remove all that "care self about encrypt/decrypt :-)
* Does this really make sense? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L165-L166 ... indicators are normally booleans only! see https://github.com/ioBroker/ioBroker/blob/master/doc/STATE_ROLES.md
* Should https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L753 be really in that if block? 
* Do I see this right that you subscribe any state you create? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L182 ? Then better to use one subscribeStates('*'); ... else it could get unperformant (or how many subscribes these are?)

Please check and adjust if needed and poke here again. thank you

Grizzelbee commented:
@Apollon77 
> Hi, sorry that it took that long, here are some review comments:

No worries. It's also not my highest prio project. ;o)


>    Please check https://github.com/Grizzelbee/ioBroker.resol/runs/4149640113?check_suite_focus=true#step:6:360

Fixed!

>    If you add common.globalDependencies with admin>=4.0.9 you can remove all that "care self about encrypt/decrypt :-)

Fixed!

> Does this really make sense? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L165-L166 ... indicators are normally booleans only! see https://github.com/ioBroker/ioBroker/blob/master/doc/STATE_ROLES.md

you're right. Switched over to "switch" since these defenately are switches.

> Should https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L753 be really in that if block?

From my point of view: Yes. It should only be executed when the config has been tested successfully and seems to be valid.

> Do I see this right that you subscribe any state you create? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L182 ? Then better to use one subscribeStates('*'); ... else it could get unperformant (or how many subscribes these are?)

Nope. These are not all states I create, but only the possible actions for the current controller. Usually much less than 10. In my case (for my controllertype) just 3.

Apollon77 commented:
Hi thank you for your reponse.

I had another look:
* makes this sense? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L168 ... a switch is a boolean and should not have "states" list (because "states" are only defined for type number ...)

Rest seems ok

Grizzelbee commented:
Hi @Apollon77 

> makes this sense? https://github.com/Grizzelbee/ioBroker.resol/blob/master/main.js#L168 ... a switch is a boolean and should not have "states" list (because "states" are only defined for type number ...)

Made my thoughts about that and came  to the conclusion that, from my point of view, it makes sense. 
In fact it's really a switch (on, auto, off) and it's states are represented by the values 0,1,2. To indicate it's functionality and make it more conveniant to the user and less documentation intensive I added these states. 
Since it can't be the only "on, auto, off"-switch ... how do others solve this issue? I'm always open to learn something new and do things better.

Apollon77 commented:
@Grizzelbee You got me wrong. By definition a role "switch" is a boolean - nothing else! "sates" are only officially supported for number states, and they can never be a switch.

Grizzelbee commented:
@Apollon77 
Okay. Got it. According to the documentation I switched over to role "level" in v0.4.3

Apollon77 committed:
Merge branch 'master' into add_resol_to_latest
